src/common/enums/preference.type.ts

```diff
@@ -0,0 +1,23 @@
+import { registerEnumType } from '@nestjs/graphql';
```

Comment:
You already have the preferences, try replacing with the following, avoiding code duplication and forcing future compatibility when enums are changed:

```suggestion
import { registerEnumType } from '@nestjs/graphql';
import { UserPreferenceType } from './user.preference.type';
import { HubPreferenceType } from './hub.preference.type';

export const PreferenceType = {
  ...UserPreferenceType,
  ...HubPreferenceType,
};

export type PreferenceType = UserPreferenceType | HubPreferenceType;

registerEnumType(PreferenceType, {
  name: 'PreferenceType',
});

```

Reply:
Feedback is incorporated in base branch